tools/iscc-sum/iscc_verify.xml

```diff
@@ -0,0 +1,147 @@
+<tool id="iscc_sum_verify" name="Verify ISCC hash" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="24.1">
+    <description>with ISCC-SUM</description>
+    <macros>
+        <import>macros.xml</import>
+        <import>creators.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <expand macro="version_command" />
+    <creator>
+        <expand macro="creators/iscc" />
+        <expand macro="creators/lco" />
+        <expand macro="creators/maartenpaul" />
+        <expand macro="creators/etzm" />
+    </creator>
+        
+    <command detect_errors="exit_code"><![CDATA[
+        GENERATED=\$(iscc-sum '$input_file' | cut -d':' -f2 | cut -d' ' -f1) &&
+        #if $expected_code_source.source_type == "text":
+            EXPECTED='$expected_code_source.expected_code_text'
+        #else:
+            ## Read file and remove all whitespace including newlines
+            EXPECTED=\$(cat '$expected_code_source.expected_code_file' | head -n 1 | tr -d '[:space:]')
+        #end if
+        &&
+        echo "File: $input_file.element_identifier" > '${output_file}' &&
+        echo "Expected: \$EXPECTED" >> '${output_file}' &&
+        echo "Generated: \$GENERATED" >> '${output_file}' &&
+        if [ "\$GENERATED" = "\$EXPECTED" ]; then
+            echo "Status: OK" >> '${output_file}';
+        else
+            echo "Status: FAILED" >> '${output_file}';
+        fi
+    ]]></command>
+    
+    <inputs>
+        <param name="input_file" type="data" format="data" label="File to verify"/>
+        <conditional name="expected_code_source">
+            <param name="source_type" type="select" label="Source of expected ISCC code">
+                <option value="text">Enter manually</option>
+                <option value="file">From file (workflow input)</option>
+            </param>
+            <when value="text">
+                <param name="expected_code_text" type="text" label="Expected ISCC code" 
+                       help="The 55-character ISCC-SUM code to verify against">
+                    <validator type="length" min="55" max="55" message="ISCC code must be exactly 55 characters"/>
+                </param>
+            </when>
+            <when value="file">
+                <param name="expected_code_file" type="data" format="txt" label="File containing expected ISCC code"
+                       help="Text file containing the ISCC code from a previous step"/>
+            </when>
+        </conditional>
```

Comment:
I see that you want to support both cases, *(i)* where the code is given explicitly by typing it into the corresponding input field, and *(ii)* where the code is given as a file. However, to reduce complexity, you might want to consider only supporting case *(i)*, because, in workflows, you can always use the [**🔧 Parse parameter value**](https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/workflow-parameters/tutorial.html#hands-on-construct-workflow-with-parameters-read-from-a-dataset) tool to pipe the contents of a file into the input field of a tool. This tool is only available in the workflow editor, not in the general tool panel.